x-pack/plugins/actions/server/lib/action_executor.ts

```diff
@@ -78,113 +79,135 @@ export class ActionExecutor {
       );
     }
 
-    const {
-      logger,
-      spaces,
-      getServices,
-      encryptedSavedObjectsClient,
-      actionTypeRegistry,
-      eventLogger,
-      preconfiguredActions,
-      getActionsClientWithRequest,
-    } = this.actionExecutorContext!;
-
-    const services = getServices(request);
-    const spaceId = spaces && spaces.getSpaceId(request);
-    const namespace = spaceId && spaceId !== 'default' ? { namespace: spaceId } : {};
-
-    const { actionTypeId, name, config, secrets } = await getActionInfo(
-      await getActionsClientWithRequest(request, source),
-      encryptedSavedObjectsClient,
-      preconfiguredActions,
-      actionId,
-      namespace.namespace
-    );
+    return withSpan(
```

Comment:
Will we eventually want to add this as a wrapper around all task type executors?

Reply:
It's only necessary if you want to group a set of async operations, so you can inspect aggregated data over it, or more easily inspect it in the trace waterfall in the APM app. I added it here for the latter reason.